### PR TITLE
Remove OpenSSL

### PR DESCRIPTION
**Pull request by 0xTim**
Removes OpenSSL and migrates calls to our own vendored BoringSSL (#14, #13).

### CODE REVIEW

**Comment by 0xTim** [comment]
@tanner0101 do you want to merge #7 and I can update this PR?

File: Sources/JWTKit/Signing/ECDSA/ECDSAKey.swift [hunk 1]
```diff
@@ -15,7 +15,7 @@ public final class ECDSAKey: OpenSSLKey {
         where Data: DataProtocol
     {
         let c = try self.load(pem: data) { bio in
-            PEM_read_bio_EC_PUBKEY(convert(bio), nil, nil, nil)
+            CJWTKitBoringSSL_PEM_read_bio_EC_PUBKEY(convert(bio), nil, nil, nil)
```
Comment (tanner0101) [inline Sources/JWTKit/Signing/ECDSA/ECDSAKey.swift]: We should be able to remove the `convert` func and any references to it. That may allow us to be a bit more type safe, too!

Comment (0xTim) [reply Sources/JWTKit/Signing/ECDSA/ECDSAKey.swift]: Done!

File: Sources/JWTKit/Utilities.swift [hunk 2]
```diff
@@ -18,25 +18,3 @@ extension UInt8 {
         return Character(".").asciiValue!
     }
 }
-
-// pointer hacks
```
Comment (tanner0101) [inline Sources/JWTKit/Utilities.swift]: ❤️

**Comment by tanner0101** [comment]
These changes are now available in [4.0.0-beta.3](https://github.com/vapor/jwt-kit/releases/tag/4.0.0-beta.3)

**Pull request merged**